### DISCUSSION TITLE
Is it possible to use a custom array index key?

### DISCUSSION THREAD

**Discussion by Zerebokep** [TanStack/query#5077] [Q&A] [answered]
Hey everyone,

I would like to know if it is somehow possible to modify they array keys which react query is using for it's cache.

For example:

<img width="294" alt="Screenshot" src="https://user-images.githubusercontent.com/15731988/222960737-eedb4475-9e00-46cb-a2a1-81cfc9f188be.png">

I would like to use the unique id from each object instead of a sequential number. Reason is that I'm using subscriptions to update the entries and I would like to update them by id instead to iterate through the whole array (which could contain thousands of entries in my project) every time.

**Comment by TkDodo** [comment, marked as answer]
the structure of data in the query cache is determined by what your QueryFunction returns. React Query doesn't dictate a structure (except for InfiniteQueries)
